### PR TITLE
Speed up CI builds by caching the pip directory

### PR DESCRIPTION
**Pull request by mdickinson**
A trivial build PR to cache pip eggs.

I plan to run the CI build twice: if this worked, the second run should be appreciably faster than that first.

### CODE REVIEW

**Comment by mdickinson** [comment]
First Travis CI build: Ran for 13 min 26 sec; Total time 19 min 31 sec

**Comment by mdickinson** [comment]
Second build was better: Ran for 8 min 54 sec; Total time 18 min 59 sec

**Comment by mdickinson** [comment]
I'll see if I can add caching for appveyor, too. Waiting for the current build to finish first.

**Comment by mdickinson** [comment]
> Waiting for the current build to finish first.

Scratch that; it was taking too long. We still expect the next Appveyor build to be slow. The one after that _should_ be faster.

**Comment by mdickinson** [comment]
The uncached Appveyor build took 20 min 38 sec. Have restarted.

**Comment by mdickinson** [comment]
Caching is now working on both Travis and Appveyor. It's not having a huge effect on Appveyor: we're down to around 17 minutes from 20 minutes, but it's better than nothing.

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,8 @@ dist: xenial
 services:
   - xvfb
 
+cache: pip
```
Comment (rahulporuri) [inline .travis.yml:6]: if im not wrong, this means `~/.cache/pip`

File: appveyor.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 build: false
 
+cache:
+  - C:\Users\appveyor\AppData\Local\pip\Cache -> appveyor.yml
```
Comment (rahulporuri) [inline appveyor.yml:4]: if i'm not wrong, this means that the cache will be invalidated if the appveyor yaml file is updated.

Comment (mdickinson) [reply appveyor.yml:4]: Yes. This gives us an easy way to invalidate the cache should we need to.

File: .travis.yml (hunk 1 above)
Comment (mdickinson) [reply .travis.yml:6]: Yes, `$HOME/.cache/pip`, according to [the documentation](https://docs.travis-ci.com/user/caching/#pip-cache)

**Pull request merged**